activerecord/test/models/company.rb

```diff
@@ -26,6 +26,9 @@ def arbitrary_method
   def private_method
     "I am Jack's innermost fears and aspirations"
   end
+
+  class SpecialCo < Company
+  end
```

Comment:
@senny unfortunately the only other subclass within the `Namespaced` module is `Client`, which [also exists outside of the module](https://github.com/rails/rails/blob/db0e649aba7135133154a35549219cffe968eaf6/activerecord/test/models/company.rb#L116).

Looking into the other issue very quickly, it appears that `find_sti_class` is working correctly:

```ruby
subclass = find_sti_class(subclass_name) # => Namespaced::Firm(id: integer, type: string, firm_id: integer, firm_name: string, name: string, client_of: integer, rating: integer, account_id: integer, description: string)
```

However, `descendants` is returning an empty array for some reason. I know very little about `ActiveSupport::DescendantsTracker`, so I'm not quite sure where to start investigating why it would be empty, particularly after forcing autoloading of `Namespaced::Firm`.